spec/vast_trackers.spec.js

```diff
@@ -223,5 +223,43 @@ describe('VASTTracker', function() {
         expect(spyEmitter).toHaveBeenCalledTimes(1);
       });
     });
+
+    describe('setProgress', () => {
+      let spyTrack;
+      let lastMock;
+      beforeEach(() => {
+        spyTrack = jest.spyOn(vastTracker, 'track');
+        vastTracker.assetDuration = 10;
+        vastTracker.setProgress(5);
+      });
+      it('should be defined', () => {
+        expect(vastTracker.setProgress).toBeDefined();
+      });
+      it('call track with progress-5', () => {
+        expect(spyTrack).toHaveBeenCalledWith('progress-5', expect.anything());
+      });
+
+      it('call track with progress-50%', () => {
+        expect(spyTrack).toHaveBeenCalledWith(
+          'progress-50%',
+          expect.anything()
+        );
+      });
+      it('should have a lastPercentage variable', () => {
+        expect(vastTracker.lastPercentage).toBeDefined();
+      });
+      it('should make the lastPercentage variable at the value -1', () => {
```

Comment:
I don't understand the goal of this unit test, why should lastPercentage be equal to -1? Also if you want to test a single feature, instead of having multiple `it` with the same title you can do multiple `expect` inside one `it`. But If you want to test the use case where we want to track 5% but detect that the last percentage tracked was 1%, meaning that previous values are missing (2%, 3%, 4%) and should have been triggered you could write:
```js
it('should also calls track for previous missing percentages', () => {
  vastTracker.lastPercentage = 1;
  expect(spyTrack.mock.calls).toContainEqual(
    ['progress-2%', expect.anything()],
    ['progress-3%', expect.anything()],
    ['progress-4%', expect.anything()]
  )
});
```